infra/auth_init.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+# Pre-provision hook to set up Azure/Entra ID app registration for FastMCP OAuth Proxy
+
+# Check if USE_FASTMCP_AUTH is enabled
+USE_FASTMCP_AUTH=$(azd env get-value USE_FASTMCP_AUTH 2>/dev/null || echo "false")
+if [ "$USE_FASTMCP_AUTH" != "true" ]; then
+    echo "Skipping auth init (USE_FASTMCP_AUTH is not enabled)"
+    exit 0
+fi
+
+echo "Setting up Azure/Entra ID app registration for FastMCP OAuth Proxy..."
```

Comment:
I hit a `KeyError: 'AZURE_TENANT_ID'` during `azd up` here. I think it is because the variable isn't exported to pre-provision hooks.

I manually added:

```
AZURE_TENANT_ID=$(azd env get-value AZURE_TENANT_ID 2>/dev/null)
if [ $? -ne 0 ]; then AZURE_TENANT_ID=""; fi
export AZURE_TENANT_ID

if [ -z "$AZURE_TENANT_ID" ]; then
    # Fallback: Extract tenant ID from azd auth token
    token_json=$(azd auth token --output json 2>/dev/null)
    if [ -n "$token_json" ]; then
        AZURE_TENANT_ID=$(python -c "import sys, json, base64; ... print(json.loads(base64.b64decode(payload).decode('utf-8'))['tid'])" <<< "$token_json")
        export AZURE_TENANT_ID
    fi
fi
```
and it worked.